src/main/java/com/blackduck/integration/sca/upload/file/FileUploaderResponse.java

```diff
@@ -0,0 +1,109 @@
+package com.blackduck.integration.sca.upload.file;
+
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.Collections;
+import java.util.Map;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+
+import com.blackduck.integration.rest.RestConstants;
+import com.blackduck.integration.rest.exception.IntegrationRestException;
+import com.blackduck.integration.rest.response.Response;
+
+public class FileUploaderResponse implements Response {
```

Comment:
Rather implementing a Response, would it make more sense to instead wrap a Response in this object? It likely would require further refactoring above in `executeUploadPart` to support however. I am mostly concerned with the amount of runtime exceptions that can occur in this class especially since we're coding around edge cases.

Reply:
Is this class only used for the IntegrationRestException case?  If so then we could rename this to IntegrationRestExceptionResponse to be more explicit.